bundles/org.openhab.binding.herzborg/src/main/java/org/openhab/binding/herzborg/internal/SerialBusHandler.java

```diff
@@ -0,0 +1,149 @@
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.herzborg.internal;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.TooManyListenersException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.io.transport.serial.PortInUseException;
+import org.openhab.core.io.transport.serial.SerialPort;
+import org.openhab.core.io.transport.serial.SerialPortEvent;
+import org.openhab.core.io.transport.serial.SerialPortEventListener;
+import org.openhab.core.io.transport.serial.SerialPortIdentifier;
+import org.openhab.core.io.transport.serial.SerialPortManager;
+import org.openhab.core.io.transport.serial.UnsupportedCommOperationException;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SerialBusHandler} implements specific handling for Herzborg serial bus,
+ * connected directly via a serial port.
+ *
+ * @author Pavel Fedin - Initial contribution
+ */
+@NonNullByDefault
+public class SerialBusHandler extends BusHandler implements SerialPortEventListener {
+    private final Logger logger = LoggerFactory.getLogger(SerialBusHandler.class);
+    private SerialPortManager serialPortManager;
+    private SerialBusConfiguration config = new SerialBusConfiguration();
+    private @Nullable SerialPort serialPort;
+
+    public SerialBusHandler(Bridge bridge, SerialPortManager portManager) {
+        super(bridge);
+        serialPortManager = portManager;
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(SerialBusConfiguration.class);
+
+        SerialPortIdentifier portIdentifier = serialPortManager.getIdentifier(config.port);
+        if (portIdentifier == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No such port: " + config.port);
+            return;
+        }
+
+        SerialPort commPort;
+        try {
+            commPort = portIdentifier.open(this.getClass().getName(), 2000);
+        } catch (PortInUseException e1) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Port " + config.port + " is in use");
+            return;
+        }
+
+        try {
+            // Herzborg serial bus operates with fixed parameters
+            commPort.setSerialPortParams(9600, SerialPort.DATABITS_8, SerialPort.STOPBITS_1, SerialPort.PARITY_NONE);
+            commPort.enableReceiveThreshold(8);
+            commPort.enableReceiveTimeout(1000);
```

Comment:
Done. However, how are communication timeouts handled if enableReceiveTimeout() is not implemented? What if there's some glitch on the serial line and one byte gets swallowed by noise? This would hand the whole binding because it would keep waiting for the missing byte forever.